ably/realtime_channel.go

```diff
@@ -114,6 +125,46 @@ func (ch *RealtimeChannels) Get(name string, options ...ChannelOption) *Realtime
 	return c
 }
 
+// GetDerived creates a new derived [ably.RealtimeChannel] object for given channel name, using the provided derive options and
+// channel options if any. Returns error if any occurs
+func (ch *RealtimeChannels) GetDerived(name string, deriveOptions DeriveOptions, options ...ChannelOption) (*RealtimeChannel, error) {
+	if deriveOptions.Filter != "" {
+		match, err := matchDerivedChannel(name)
+		if err != nil {
+			return nil, err
+		}
+		filter := url.PathEscape(deriveOptions.Filter)
```

Comment:
Should this be `url.QueryEscape`? eg `url.PathEscape` doesn't encode ':' characters

Reply:
`url.QueryEscape` causes an `invalid filter expression code` error. An alternative is to use [url.Parse](https://pkg.go.dev/net/url#URL.Parse).

Reply:
Ah ok `QueryEscape` encodes spaces as `+`, which is valid for a URL but we can't decode it in JS to get back the original string...

Maybe for now we stick with `PathEscape` but replace `:` with `%3A`? The characters that we need to make sure are properly escaped are `:` (as its used to separate channel scopes) - and `?` and `]` (which both seem ok).

Eg using a filter 'foo == `bar:car`' will fail as an invalid channel name

In hindsight using URL encoding was probably the wrong choice (@mschristensen maybe in the production release we switch to base64 or something instead to avoid any differences between language encodings?)

Reply:
@mohyour if its ok with you will update update realtime to expect base64 instead which should be rolled out to sandbox next week (https://github.com/ably/realtime/pull/4835)

Reply:
@mohyour I think we should make this change now and merge once tests pass when the realtime change goes to sandbox :) If you want to test your implementation locally, sync with @andydunstall who should be able to spin up a dev cluster with the change that you can run the tests against :)